MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs

```diff
@@ -550,24 +550,14 @@ private static void HandleReplaceCube(GameSession session, PacketReader packet)
 
         private static void HandlePickup(GameSession session, PacketReader packet)
         {
-            byte[] coords = packet.Read(3);
+            CoordB coords = packet.Read<CoordB>();
 
-            // Convert to signed byte array
-            sbyte[] sCoords = Array.ConvertAll(coords, b => unchecked((sbyte) b));
-            // Default to rainbow tree
-            int weaponId = 18000004;
-
-            // Find matching mapObject
-            foreach (MapObject mapObject in MapEntityStorage.GetObjects(session.Player.MapId))
+            int weaponId = MapEntityStorage.GetWeaponObjectItemId(session.Player.MapId, coords);
```

Comment:
Shouldn't you check if the cube is in the Field State?

Reply:
Nope. These objects aren't loaded in the field.